tools/flaky-test-retryer/github_commenter.go

```diff
@@ -105,7 +158,14 @@ func (gc *GithubClient) getOldComment(org, repo string, pull int) (*github.Issue
 	}
 	var match *github.IssueComment
 	for _, comment := range comments {
-		found, err := regexp.Match(identifier, []byte(comment.GetBody()))
+		body := []byte(comment.GetBody())
+		found, err := regexp.Match(legacyIdentifier, body)
+		if err != nil || !found {
+			if testNameFromComment := regexp.MustCompile(reTestIdentifier).FindStringSubmatch(comment.GetBody()); len(testNameFromComment) >= 2 {
```

Comment:
```suggestion
			if testNameFromComment := regexp.MustCompile(reTestIdentifier).FindStringSubmatch(body); len(testNameFromComment) >= 2 {
```

Reply:
this doesn't work as it's expecting string instead of byte slice

Reply:
we can use
```
body = comment.GetBody()
found, err := regexp.Match(legacyIdentifier, []byte(body))
if testNameFromComment := regexp.MustCompile(reTestIdentifier).FindStringSubmatch(body); len(testNameFromComment) >= 2 {
```

There are multiple GetBody() calls here. Wondering if we can only call it when necessary :confused:

Reply:
done